sr_wear_logger/src/sr_wear_logger.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python
+#
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+import rospkg
+import os
+import yaml
+import roslaunch
+import requests
+import shutil
+import datetime
+from sensor_msgs.msg import JointState
+from sr_utilities_common.aws_manager import AWS_Manager
+
+
+class WearLogger():
+
+    def __init__(self):
+        self.aws_manager = AWS_Manager()
+        self._first_run = True
+        self._previous_values = {}
+        self._current_values = {}
+        self._threshold = 0.0175
+        self._hand_serial = rospy.get_param("~hand_serial")
+        self._log_file_path = rospkg.RosPack().get_path('sr_wear_logger') + "/" + self._hand_serial + "/"
+        self._log_file_name = "wear_data.yaml"
+        self._init_log()
+
+        if self._hand_serial != "":
+            rospy.Timer(rospy.Duration(2), self._save_data_localy)
+            rospy.Timer(rospy.Duration(10), self._upload_to_AWS)
+
+        rospy.Subscriber('/joint_states', JointState, self.callback)
+
+    def _extract_hand_data(self, msg):
+        hand_data = dict(zip(msg.name, msg.position))
+        for key in hand_data.keys():
+            if key.startswith("ra") or key.startswith("la"):
+                hand_data.pop(key, None)
+        return hand_data
+
+    def callback(self, msg):
+        hand_data = self._extract_hand_data(msg)
+        updates = dict.fromkeys(hand_data.keys(), 0)
+        if self._first_run:
+            self._previous_values = hand_data
+            self._first_run = False
+        for key, value in hand_data.items():
+            value_difference = abs(self._previous_values[key] - value)
+            if value_difference > self._threshold:
+                self._current_values[key] += round(value_difference, 5)
+        self._previous_values = hand_data
+
+    def _get_latest_file_name_from_AWS(self):
+
+        files = self.aws_manager.get_bucket_structure_with_prefix("shadowrobot.benchmarks", self._hand_serial)
+        if files is None:
```

Comment:
should this return an empty string? what happens if an empty string is actually returned? we should consider using an exception like this:
```
try:
except None as e:
``` 
unless im missing something.

Reply:
get_bucket_structure_with_prefix will return either None (in case there are no files within a path with given prefix), or the list of files otherwise